install_config/upgrading/blue_green_deployments.adoc

```diff
@@ -0,0 +1,159 @@
+excess capacity to facilitate such an approach.
+
+ifdef::openshift-enterprise[]
+Another lesser challenge is that the administrator must temporarily share the
+Red Hat software entitlements between the blue-green deployments or provide
+access to the installation content by means of a system such as Red Hat
+Satellite. This can be accomplished by sharing the consumer ID from the previous
+host.
+
+. On the old host:
++
+----
+# subscription-manager identity | grep system
+system identity: 6699375b-06db-48c4-941e-689efd6ce3aa
+----
+
+. On the new host:
++
+----
+# subscription-manager register --consumerid=6699375b-06db-48c4-941e-689efd6ce3aa
+----
+
+[IMPORTANT]
+====
+After a successful deployment, remember to unregister the old host with
+`subscription-manager clean` to prevent the environment from being out of
+compliance.
+====
+
+endif::openshift-enterprise[]
+
+[[blue-green-deployments-preparing-for-upgrade]]
+== Preparing for Upgrade
+
+xref:../../install_config/upgrading/manual_upgrades.adoc#upgrading-masters[After
+the master and etcd servers have been upgraded], you must ensure that your
+current production nodes are labeled either blue or green. In this example, the
+current installation will be blue and the new environment will be green. On each
+production node in your current installation:
+
+----
+$ oc label --all nodes color=blue
+----
+
+In the case of nodes requiring the uptime guarantees of a blue-green deployment,
+the `-l` flag can be used to match a subset of the environment using a selector.
+
+Next, xref:../install/advanced_install.adoc#adding-nodes-advanced[the green
+environment] should be created for any nodes that are to be replaced. Ansible
+can apply the `*color=green*` label using the `*openshift_node_labels*` variable
+for each node.
+
+In order to delay workload scheduling until the nodes are
+xref:../../architecture/infrastructure_components/kubernetes_infrastructure.adoc#node[healthy],
+be sure to set the `*openshift_schedulable=false*` variable.
+
+After the green nodes are in *Ready* state, they can be made schedulable. Then,
+blue nodes are disabled so that no new pods are run on them:
+
+----
+# oadm manage-node --schedulable=true --selector=color=green
+# oadm manage-node --schedulable=false --selector=color=blue
+----
+
+A common practice is to scale the registry and router pods until they are
+migrated to the green nodes. For these pods, a _canary_ deployment approach is
+commonly used. Scaling them up will make them immediately active on the new
+nodes. Pointing the deployment configuration to the new image initiates a
+rolling update. However, because of node anti-affinity, and the fact that the
+blue nodes are still unschedulable, the deployments to the old nodes will fail.
+At this point, the registry and router deployments can be scaled down to the
+original number of pods. At any given point, the original number of pods is
+still available so no capacity is lost.
```

Comment:
Is there a way the tasks in this "Preparing for Upgrade" section can be broken up into numbered steps? Seems like that would be easier to grok at a glance.